docs/layout-of-source-files.rst

```diff
@@ -7,6 +7,40 @@ Source files can contain an arbitrary number of
 :ref:`pragma directives<pragma>` and
 :ref:`struct<structs>` and :ref:`enum<enums>` definitions.
 
+.. index:: ! license, spdx
+
+SPDX License Identifier
+=======================
+
+Trust in smart contract can be better established if their source code
+is available. Since making source code available always touches on legal problems
+with regards to copyright, the Solidity compiler encouranges the use
+of machine-readable `SPDX license identifiers <https://spdx.org>`_.
+Every source file should start with a comment indicating its license:
+
+::
+
+    // SPDX-License-Identifier: MIT
+
+The compiler does not validate that the license is part of the
+`list allowed by SPDX <https://spdx.org/licenses/>`_, but
+it does include the supplied string in the `bytecode metadata <metadata>`_.
+
+If you do not want to specify a license or if the source code is
+not open-source, please use the special value ``UNLICENSED``.
```

Comment:
Should we mention this is also the suggested way by npm?

Reply:
I don't think this is needed.

Reply:
I just saw that there is a spdx license identifier [`UNLICENSE`]( https://spdx.org/licenses/Unlicense.html). I'm wondering whether `UNLICENSE` vs `UNLICENSED` can create confusions.

```json
    {
      "reference": "./Unlicense.html",
      "isDeprecatedLicenseId": false,
      "isFsfLibre": true,
      "detailsUrl": "http://spdx.org/licenses/Unlicense.json",
      "referenceNumber": "179",
      "name": "The Unlicense",
      "licenseId": "Unlicense",
      "seeAlso": [
        "https://unlicense.org/"
      ],
      "isOsiApproved": false
    }
```

Reply:
This is also recommended by npm, I don't think we should just make an arbitrary new suggestion.